src/LinearCurve.sol

```diff
@@ -30,80 +30,64 @@ import {BaseCurve} from "./BaseCurve.sol";
  *      versions.
  */
 contract LinearCurve is BaseCurve {
-    /// @dev UD60x18 Max
-    uint256 public immutable MAX_SHARES = type(uint256).max / 1e18;
+    /// @dev Maximum number of shares that can be handled by the curve.
+    uint256 public constant MAX_SHARES = type(uint256).max;
 
-    /// @dev UD60x18 Max
-    uint256 public immutable MAX_ASSETS = type(uint256).max / 1e18;
+    /// @dev Maximum number of assets that can be handled by the curve.
+    uint256 public constant MAX_ASSETS = type(uint256).max;
 
     /// @notice Constructor for the Linear Curve.
     /// @param _name The name of the curve.
     constructor(string memory _name) BaseCurve(_name) {}
 
     /// @inheritdoc BaseCurve
     /// @notice Computes the 1:1 relationship between assets <--> shares.
-    function previewDeposit(uint256 assets, uint256 totalAssets, uint256 totalShares)
```

Comment:
3.20 - Functions in Curve Contracts Could Be external Instead of public

Made all of these `external`